f/frizzle/kobo/kobo_responses.py

```diff
@@ -34,92 +34,72 @@ def main(
-                        if response.status_code == 200:
-                            file_name = attachment["filename"]
-                            # ie. datalake/{dataset_id}/attachments/{files}
-                            save_path = os.path.join(
-                                attachment_root,
-                                my_forms["dataset_id"],
-                                "attachments",
-                                os.path.basename(file_name),
-                            )
-                            os.makedirs(os.path.dirname(save_path), exist_ok=True)
-                            with open(save_path, "wb") as file:
-                                file.write(response.content)
-                            logger.debug("Download completed: " + attachment["download_url"])
-                        else:
-                            # TODO: add retries
-                            logger.error("Failed downloading attachments.")
-        submissions = current_form_data
-        logger.info(
-            f"Form {index + 1} (ID: {form_id}): Fetched {len(submissions)} submission(s)."
-        )
-    return form_data
+    # Next download the form questions & metadata
+    # FIXME: need to paginate. Maximum results per page is 30000.
+    response = requests.get(data_uri, headers=headers)
+    response.raise_for_status()
+
+    all_submissions = response.json()["results"]
+
+    for submission in all_submissions:
+        # Inject the form name/id to each submission.
+        # Thus these fields will be added as columns along with all the other submission info later on.
+        submission["dataset_name"] = form_name
+        submission["dataset_id"] = dataset_id
+
+        # If the result has attachments, download them.
+        if "_attachments" in submission:
+            for attachment in submission["_attachments"]:
+                if "download_url" in attachment:
+                    response = requests.get(attachment["download_url"], headers=headers)
+                    if response.status_code == 200:
+                        file_name = attachment["filename"]
+                        # ie. datalake/{dataset_id}/attachments/{files}
+                        save_path = os.path.join(
+                            attachment_root,
+                            dataset_id,
+                            "attachments",
+                            os.path.basename(file_name),
+                        )
+                        os.makedirs(os.path.dirname(save_path), exist_ok=True)
+                        with open(save_path, "wb") as file:
+                            file.write(response.content)
+                        logger.debug(
+                            "Download completed: " + attachment["download_url"]
+                        )
+                    else:
+                        # TODO: add retries
+                        logger.error("Failed downloading attachments.")
+
+    logger.info(
```

Comment:
Currently, we differ on where we log success messages for the different steps in the data flow. In comapeo (following suit from the previous convention in kobo), [they are in `main`](https://github.com/ConservationMetrics/gc-scripts-hub/blob/d38924208895f4a7536ecfaf807eadb3d0ff3c23/f/frizzle/comapeo/comapeo_observations.py#L40-L56). Here I see you've opted to place the log in the function itself, which I agree is the better choice. We could consider doing the same for the DBWriter.

Anyway, I know that's not the point here. We can address this (and other kobo script modifications) in another PR.